db/dbformat.h

```diff
@@ -214,7 +214,7 @@ class InternalKeyComparator
   virtual void FindShortSuccessor(std::string* key) const override;
 
   const Comparator* user_comparator() const {
-    return user_comparator_.user_comparator();
+    return &user_comparator_;
```

Comment:
There seems to be a lot of caller to this functions. Are we fixing accounts for all of them, or is there a chance we overcount somewhere?

Reply:
I think about it again. Can we create a new function that returns the wrapper instead? In that case, the user of the function will be explicit about whether the comparisons are counted underlying or not. And it can also potentially avoid one virtual function call.

Reply:
Do you mind helping me understand what kind of comparison should not be counted when `PerfContext` is enabled? I am not that familiar with this wrapper so am still naively hoping we can land in one of the following simple situations:

1. the wrapper is cheap and we allow it to count all comparisons
2. the wrapper is expensive so we get rid of it entirely

Reply:
Never mind, I saw there's an existing pattern to follow (call `UserComparatorWrapper(rep_->internal_comparator.user_comparator()).Compare()`). Will follow that pattern.